### PR TITLE
Feature: Action Receipt Invalidator

### PR DESCRIPTION
**Pull request by Bexanderthebex**
### 🤔 Why?

- We recently added a way to store app_receipt_actions into our postgres instance
- However, currently it is unbounded and is concerning in the point of view of the application as it is not the initial goal of the project
- In order to combat this, we will want to add a way to put a bound on this so that we don't pay for unnecessary data on our side

### 🛠 What I changed:

- Added a cron job that will run once everyday to remove transactions that we don't intend to query

### 🚦 Functional Testing Results:

- Tested the query locally and made sure that only the records we want to remove are affected

### CODE REVIEW

File: src/crons/appActionReceipts.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { CronJob } from './CronJob';
+import { DAY, MINUTE } from '../utils/constants';
+
+type AppActionReceiptsSpec = {
+  localCachePool: DatabasePool;
+  indexerCachepool: DatabasePool;
+  environment: Record<string, string>;
+};
+
+interface actionReceiptActionProps {
```
Comment (encody) [inline src/crons/appActionReceipts.ts:11]: CapitalizedCamelCase for types

File: src/crons/appActionReceipts.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          receipt_included_in_block_timestamp > ${startEpoch} and receipt_included_in_block_timestamp <= ${endEpoch}
+      `);
+
+      if (!actionReceipts) {
+        return;
+      }
+
+      const actionReceiptValues = actionReceipts.map(a => {
+        const v = { ...a, args: JSON.stringify(a.args) };
+        return Object.values(v);
```
Comment (encody) [inline src/crons/appActionReceipts.ts:77]: Arbitrary order

File: src/crons/appActionReceipts.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      }) as readonly any[];
```
Comment (encody) [inline src/crons/appActionReceipts.ts:78]: Try to avoid `any` as much as possible

File: src/index.ts [hunk 4]
```diff
@@ -36,6 +36,7 @@ const indexerPool = createPool(indexerDatabaseString, {
 process.on('exit', async () => {
   try {
     await Promise.all([await cachePool.end(), await indexerPool.end()]);
+    console.log('succesffuly closed database connections');
```
Comment (encody) [inline src/index.ts:39]: spell check

**Comment by encody** [comment]
Build failing? I'm going to try pushing a trivial commit and see if it was just a fluke, because nothing about the Docker config should have changed.

**Comment by encody** [comment]
Weird.

**Comment by Bexanderthebex** [comment]
Oh yeah, it was a fluke I think that was related to a package not downloaded because the package repo was down or something

**Pull request merged**